src/main/java/appeng/recipes/game/WirelessTerminalRecipe.java

```diff
@@ -0,0 +1,162 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Applied Energistics 2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Applied Energistics 2.  If not, see <http://www.gnu.org/licenses/lgpl>.
+ */
+
+package appeng.recipes.game;
+
+
+import java.util.ArrayList;
+import javax.annotation.Nullable;
+
+import net.minecraft.init.Items;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.inventory.InventoryCrafting;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.world.World;
+import net.minecraftforge.oredict.OreDictionary;
+
+import appeng.api.AEApi;
+import appeng.api.definitions.IDefinitions;
+import appeng.api.util.AEColor;
+
+
+public class WirelessTerminalRecipe implements IRecipe
+{
+
+    private final Item terminal;
+    private final ItemStack waterBucket;
+
+    private static final ArrayList<String> dyes = new ArrayList<String>();
+    static
+    {
+        dyes.add( "dyeBlack" );
+        dyes.add( "dyeBlue" );
+        dyes.add( "dyeBrown" );
+        dyes.add( "dyeCyan" );
+        dyes.add( "dyeGray" );
+        dyes.add( "dyeGreen" );
+        dyes.add( "dyeLightBlue" );
+        dyes.add( "dyeLightGray" );
+        dyes.add( "dyeLime" );
+        dyes.add( "dyeMagenta" );
+        dyes.add( "dyeOrange" );
+        dyes.add( "dyePink" );
+        dyes.add( "dyePurple" );
+        dyes.add( "dyeRed" );
+        dyes.add( "dyeWhite" );
+        dyes.add( "dyeYellow" );
+    }
+
+    public WirelessTerminalRecipe()
+    {
+        final IDefinitions definitions = AEApi.instance().definitions();
+        terminal = definitions.items().wirelessTerminal().item( AEColor.Black ); //pick a random color, items don't even have NBT data so it's ignored
+        waterBucket = new ItemStack( Items.water_bucket, 1 );
+    }
+
+    @Override
+    public boolean matches( InventoryCrafting inv, World w )
+    {
+        return this.getOutput( inv, false ) != null;
+    }
+
+    @Nullable
+    private ItemStack getOutput( final IInventory inv, final boolean createTerminal )
+    {
+        ItemStack inputTerminal = null;
+        String color = null;
+        int numIngredients = 0;
+        for( int i = 0; i < 9; i++ )
+        {
+            //If the slot is empty, ignore it
+            if( inv.getStackInSlot( i ) != null )
+            {
+                numIngredients++;
+                //if it contains a terminal, return that terminal
+                if(inv.getStackInSlot( i ).getItem().equals( this.terminal ) )
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) '(' is not followed by whitespace. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)